back-end/e-learning/app/Http/Controllers/UserAnswerController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Collection;
+
+use App\Models\UserAnswer;
+use App\Models\UserCategory;
+use App\Models\Choice;
+
+class UserAnswerController extends Controller
+{
+    public function saveAnswer($category_id, $word_id, $choice_id)
+    {
+        $user_category = UserCategory::where([
+            ['user_id', auth()->user()->id], ['category_id', $category_id]
+        ])->first();
+
+        if (!is_null($user_category))
+        {
+            $correct_answer = Choice::where([
```

Comment:
you can make a function in Word model to return the correct answer instead of doing a manual where statement here...

```
// Word.php

function getCorrectChoice()
{
    $this->choices()->where('is_correct_answer', 1)->get();
}
```